### PR TITLE
Prusti user guide

### PR DESCRIPTION
**Pull request by Aurel300**
[**Link to rendered preview**](https://media.thenet.sk/aca/prusti-user-guide) (as of f9dd69f)

Here is a guide that should replace the wiki tutorial. I tried to document our more recent features too. Let me know if I should add something.

### CODE REVIEW

**Comment by fpoli** [comment]
"When using the extension, ensure that ... is uses nightly" — I would add something like "If you wish to use the latest development version" somewhere.

**Comment by fpoli** [comment]
In my opinion Prusti is also meant to be used from the command line by the users; not just via the IDE. It should be simple (download the release, then launch `prusti-rustc --edition=2018 my/program.rs`), but we haven't written a README yet.

The Installation page could have two sections: Prusti Assistant and the precompiled binaries:
* For Prusti Assistant I would put as few instructions as possible, even less than what is currently there. I would rather suggest to follow the README of the extension (which we could extend).
* For the procompiled binaries, I would mention that they can be downloaded from https://github.com/viperproject/prusti-dev/releases. Those marked as "release" are meant to be a bit more stable.

File: user-guide/src/basic.md [hunk 1]
```diff
@@ -3,14 +3,64 @@
...
+        b
+    }
+}
+```
+
+When pasting the above code into a Rust file and then verifying it with Prusti (as outlined at the top of the page), Prusti should report that verification succeeded.
+This tells us that
+
+1. the file consists of valid Rust code that can be compiled successfuly, and
+2. no execution reaches an explicit call of the [`panic!()`](https://doc.rust-lang.org/std/macro.panic.html) macro.
```
Comment (Aurel300) [inline user-guide/src/basic.md]: I guess "explicit" is misleading, since a panic can be caused e.g. by an assertion failure.

**Comment by Aurel300** [comment]
https://viperproject.github.io/prusti-dev/user-guide/

**Pull request merged**